src/_actions/TradeActions.js

```diff
@@ -61,10 +61,24 @@ export const setQuickTradeField = (symbol, tradeType, field, value) => ({
     value,
 });
 
-export const createTrade = symbol => ({
-    type: types.CREATE_TRADE,
-    symbol,
-});
+export const createTrade = symbol =>
+    (dispatch, getState) => {
+        const activeIdx = getState().workspace.get('activeTradeIndex');
+        const symbolUsed = symbol || getState().trades.get(activeIdx).toJS().symbol;
+        const contractExist = getState().tradingOptions.get(symbolUsed);
+        const ticksExist = getState().ticks.get(symbolUsed);
+
+        const contractP = !!contractExist ? Promise.resolve() : LiveData.api.getContractsForSymbol(symbolUsed);
+        const ticksP = !!ticksExist ? Promise.resolve() : LiveData.api.subscribeToTick(symbolUsed);
+
+        Promise.all([contractP, ticksP])
+            .then(
+                v => {
+                    console.log(v);
+                    dispatch({ type: types.CREATE_TRADE, symbol: symbolUsed });
```

Comment:
Not sure if such coding style are due to coming from classical OOP >.<
The react push and update paradigm makes it difficult to store such logic

As for robustness, I believe it's fine, it will break of course if we change any of those state related

Any suggestion?